### DISCUSSION TITLE
Merge (collapse) multiple underscores and dashes in the identifiers

### DISCUSSION THREAD

**Discussion by Denis535** [stylus/stylus#2846] [Ideas]
I want to write all my variables in column:
```
theme--color-------------light--hover = white
theme--background-color--dark---hover = white
theme--border-color------light--hover = white
theme--border-radius                  = 0px
```
Instead of:
```
theme--color--light--hover           = white
theme--background-color--dark--hover = white
theme--border-color--light--hover    = white
theme--border-radius                 = 0px
```

So, I want Stylus to merge multiple "_", "-" into "__", "--".
I can achieve this by creating proxy for ```Ident```:
```
Stylus.nodes.Ident = new Proxy(Stylus.nodes.Ident, {{
    construct: function(target, args) {{
        args[0] = args[0].replaceAll(/(__)(_*)/g, '__').replaceAll(/(--)(-*)/g, '--');
        return new target(...args);
    }}
}});
```
But native support would be better.

**Comment by vendethiel** [comment]
What the fuck

**Comment by slightlyfaulty** [comment]
This is a very bad idea for many reasons. Like breaking IDE completion and navigation for variables. Not to mention being unintuitive to future developers looking at your code. All just to make it subjectively more visually appealing for yourself.

If you're concerned about readability and nice alignment, I'd suggest something like this instead:

```stylus
theme = {
    light: {
        color:            white
        background-color: white
        border-color:     white
    },
    dark: {
        //...
    },
}

.text
    color theme.light.color
```

Ideally you'd just be declaring your theme values like this and using [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) throughout your styling.

**Reply by Denis535** [reply]
I used object:
```
theme = {
    color: {}
    background-color: {}
    background-image {}
    border-color: {}
    border-radius: {}
}
```
But it's better to use just variables.
And it's much easier for me when all names are nicely aligned.

**Reply by slightlyfaulty** [reply]
That's fine but understand that when you're coding just for yourself it doesn't matter, but when you're coding to share with others it's always best to follow conventions and best practices, even if it might not be exactly what suits you the best.

**Reply by Denis535** [reply]
There was also problem that this assignment did not work:
```theme[color]['light'][hover] = value``` // it does nothing
I was forced to manually create each object:
```
theme[color] = {}
theme[color]['light'] = {}
theme[color]['light'][hover] = value
```
So, I was forced to use variables.
Also when keys are rgba or functions it also did not work. So, ```theme[component] = value``` will do nothing.
